From 94e775bda3a987eb2f949b026a5ac818ab61e4a1 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Tue, 5 Sep 2017 17:56:36 -0400
Subject: [PATCH 3821/5855] drm/amdgpu: fix userptr put_page handling

Follow up for Christian's change with the same headline, adapting
KFD's userptr handling.

Change-Id: I58d88345ce9b73298a483fa1c1c2d14c03b583a2
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index f6d50a1..1273522 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -529,8 +529,7 @@ static int init_user_pages(struct kgd_mem *mem, struct mm_struct *mm,
 		goto free_out;
 	}
 
-	memcpy(bo->tbo.ttm->pages, mem->user_pages,
-	       sizeof(struct page *) * bo->tbo.ttm->num_pages);
+	amdgpu_ttm_tt_set_user_pages(bo->tbo.ttm, mem->user_pages);
 
 	ret = amdgpu_bo_reserve(bo, true);
 	if (ret) {
@@ -2099,8 +2098,8 @@ static int validate_invalid_user_pages(struct amdkfd_process_info *process_info)
 
 		/* Copy pages array and validate the BO if we got user pages */
 		if (mem->user_pages[0]) {
-			memcpy(bo->tbo.ttm->pages, mem->user_pages,
-			       sizeof(struct page *) * bo->tbo.ttm->num_pages);
+			amdgpu_ttm_tt_set_user_pages(bo->tbo.ttm,
+						     mem->user_pages);
 			amdgpu_ttm_placement_from_domain(bo, mem->domain);
 			ret = ttm_bo_validate(&bo->tbo, &bo->placement,
 					      false, false);
-- 
2.7.4

